src/utils/query.js

```diff
@@ -13,70 +13,20 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
-import liskAPIInstance from './api';
-
-class Query {
-	constructor() {
-		this.client = liskAPIInstance;
-		this.handlers = {
-			account: (account, options) => this.getAccount(account, options),
-			block: (block, options) => this.getBlock(block, options),
-			delegate: (delegate, options) => this.getDelegate(delegate, options),
-			transaction: (transaction, options) =>
-				this.getTransaction(transaction, options),
-		};
-
-		this.testnetCache = null;
-	}
-
-	cacheTestnetSetting() {
-		this.testnetCache = this.client.testnet;
-	}
-
-	resetTestnetSetting() {
-		this.client.setTestnet(this.testnetCache);
-		this.testnetCache = null;
-	}
-
-	sendRequest(endpoint, parameters, { testnet } = {}) {
-		const overrideTestnetSetting =
-			typeof testnet === 'boolean' && this.client.testnet !== testnet;
-
-		const handleSuccess = async result => {
-			if (overrideTestnetSetting) this.resetTestnetSetting();
-			if (!result.success) throw new Error(result.message);
-			return result;
-		};
-		const handleFailure = async result => {
-			if (overrideTestnetSetting) this.resetTestnetSetting();
-			throw result;
-		};
-
-		if (overrideTestnetSetting) {
-			this.cacheTestnetSetting();
-			this.client.setTestnet(testnet);
-		}
-
-		return this.client
-			.sendRequest(endpoint, parameters)
-			.then(handleSuccess, handleFailure);
-	}
-
-	getBlock(input, options) {
-		return this.sendRequest('blocks/get', { id: input }, options);
-	}
-
-	getAccount(input, options) {
-		return this.sendRequest('accounts', { address: input }, options);
-	}
-
-	getTransaction(input, options) {
-		return this.sendRequest('transactions/get', { id: input }, options);
-	}
-
-	getDelegate(input, options) {
-		return this.sendRequest('delegates/get', { username: input }, options);
-	}
-}
-
-export default new Query();
+import getAPIClient from './api';
+
+export default (endpoint, parameters, { testnet } = {}) =>
+	// prettier-ignore
+	getAPIClient(testnet)[endpoint].get(parameters)
+		.then(res => {
+			if (res.data) {
+				if (Array.isArray(res.data)) {
+					if (res.data.length === 0) {
+						throw new Error('Data was not found with specified parameters.');
```

Comment:
This is a pretty obscure error message. Slightly better would be `No results were found with the specified parameters.` Can we customise it for the endpoint and/or parameters though?

Reply:
I mostly object to the word `data`.

Reply:
maybe instead of data, we use endpoint?
like
```
lisky> get delegate 123
╔═══════════════════════════════════════════════════════════════════╗
║ error                                                             ║
╟───────────────────────────────────────────────────────────────────╢
║ Could not get: delegates was not found with specified parameters. ║
╚═══════════════════════════════════════════════════════════════════╝
```

Reply:
`Could not get: delegates not found using specified endpoints.`?

Reply:
Or `no delegates found using specified endpoints`? They're reliably plural, right?

Reply:
Most of them are plural except `node` endpoint, which we don't have now on commander,
so referring as plural should be ok.
But I think the error message should be refer to parameters or options? rather than end points I think.
- user don't think about endpoint at this point
- event with the correct endpoint, it's possible to get no result depending on the paramter

Reply:
Sorry, I meant `parameters`, not `endpoints`.